crates/bevy_render/src/view/window/cursor.rs

```diff
@@ -73,12 +74,22 @@ pub enum CustomCursor {
 
 pub fn update_cursors(
     mut commands: Commands,
-    mut windows: Query<(Entity, Ref<CursorIcon>), With<Window>>,
+    windows: Query<(Entity, Ref<CursorIcon>), With<Window>>,
+    mut icon_removed: RemovedComponents<CursorIcon>,
```

Comment:
'RemovedComponents` is on the chopping block. Could you replace this with an `OnRemove` observer?